include/aws/http/connection.h

```diff
@@ -182,6 +185,17 @@ struct aws_http2_connection_options {
      */
     size_t max_closed_streams;
 
+    /**
+     * Optional
+     * The customized limitation of the buffer size for http2 connection.
+     * A default number is set by AWS_HTTP2_CONNECTION_OPTIONS_INIT to UINT32_MAX.
+     *
+     * Affects the debug data from the received GOAWAY frame.
+     * If the data from GOAWAY frame is larger than the limitation, whole debug data
+     * will be ignored.
+     */
+    uint32_t buffer_limits;
```

Comment:
yeah, I agree it's best to set the default to 0, just thought about if user wants to set it to zero to disable any buffer... But, it's probably just over thinking.

Reply:
I agree that it seems weird to prevent a buffer of size 0, but someone who wants to save space can always just set it to 1.

I just think the benefit of using 0 for defaults outweighs the benefits of saving a 1 byte allocation.
If we find that we **need** to support a value of zero in the future, we can always add another property like `bool disable_goaway_debug_data`. That's what we did with window_size. We did need to support 0, so we have an additional bool for turning the whole feature on/off (manual_window_management).